pgjdbc/src/main/java/org/postgresql/PGConnection.java

```diff
@@ -21,6 +22,22 @@
  * returned by the PostgreSQL driver implement PGConnection.
  */
 public interface PGConnection {
+
+  /**
+   * Creates an {@link Array} wrapping <i>elements</i>. This is similar to
+   * java.sql.Connection#createArrayOf(String, Object[]), but also provides
+   * support for primitive arrays.
+   *
+   * @param typeName
+   *          The SQL name of the type to map the <i>elements</i> to.
+   * @param elements
+   *          The array of objects to map.
+   * @return An {@link Array} wrapping <i>elements</i>.
+   * @throws SQLException If for some reason the array cannot be created.
+   * @see java.sql.Connection#createArrayOf(String, Object[])
+   */
+  Array createArrayOf(String typeName, Object elements) throws SQLException;
```

Comment:
Can `Object elements` be null?

Reply:
No it cannot.
Similar to how createArrayOf(String typeName, Object[] elements) cannot take null elements either.

Reply:
@bokken , here's Java 8 documentation:

https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#createArrayOf-java.lang.String-java.lang.Object:A-

>SQLException - if a database error occurs, the JDBC type is not appropriate for the typeName and the conversion is not supported, the typeName is null or this method is called on a closed connection

It does forbid `null` for `typeName`, however I do not see how it forbids `null` for `elements`

The added `PGConnection.createArrayOf(String, Object)` is a public API, and I think nullability documentation is a must.

PS. Sorry for the long-standing discussion, however we'd better be picky when adding new APIs.

Reply:
I was speaking to the `PGConnection` implementation of `createArrayOf`, which also throws a `NullPointerException` if given `null` `elements`.
As this was not documented there, I did not document here.

https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java#L1149

Reply:
Perhaps both methods should return `null` if `elements` is `null`?
I was simply expecting/providing symmetry with the existing implementation.

Reply:
`NullPointerException if given null elements` might be an existing bug.

Reply:
Perhaps both methods should return `null`?
I was simply expecting/providing symmetry in the implementations between the 2 methods.

Reply:
It appears that it is possible to create a PGArray representing a `null` array. Perhaps that is the solution?

Reply:
So it is possible to create a null array in postgres so seems reasonable to allow it 
```
select null::int[];
 int4
------
```